CONTRIBUTING.md

```diff
@@ -0,0 +1,23 @@
+# Contributing
+
+All contributions are welcome to this project!
+
+The general process for contributing code is as follows:
+
+1. Open an issue describing the desired feature, bug fix, or improvement you'd like to see before making any changes.
+2. Assign the ticket to yourself, and/or leave a comment on the issue that you would like to work on it.
+3. Fork the repository and make the necessary changes on a branch identified by the issue (e.g. `gh-##`). This ensures the branch is linked to the corresponding issue.
+4. Open a [pull request](https://help.github.com/en/articles/about-pull-requests) for the maintainers of the repository to review.
+5. Be sure to add the maintainers to the pull request as reviewers to ensure a speedy review.
+6. Upon reviewing the changes, your change may be:
+    * merged, requiring no further changes on your part.
+    * annotated with a set of changes that must be made before the change will be merged.
+
+## Unit Tests
+
+All changes submitted must include updated unit tests (excluding documentation and configuration changes).
+In order for your change to be accepted, the [TravisCI](https://travis-ci.com/indeedeng/cluster-preset) pipeline must complete successfully.
```

Comment:
Looking through the existing repo it doesn't look like there are many unit tests for the other parts of the code, and I don't think adding test here will kick off the travis pipeline. We could probably keep the unit tests requirement but drop the travis pipeline sentence.